ci/cscs-ci.yml

```diff
@@ -59,60 +59,72 @@ stages:
     BASE_IMAGE: jfrog.svc.cscs.ch/dockerhub/nvidia/cuda:${CUDA_VERSION}-devel-ubuntu${UBUNTU_VERSION}
     EXTRA_UV_SYNC_ARGS: "--extra cuda12"
 
-# TODO: rocm steps are in draft state for now to show how to add support in the future
-# .build_extra_rocm:
-#   variables:
-#     # jfrog.svc.cscs.ch/dockerhub/rocm is the cached version of docker.io/rocm
-#     BASE_IMAGE: jfrog.svc.cscs.ch/dockerhub/rocm/dev-ubuntu-${UBUNTU_VERSION}:${ROCM_VERSION}-complete
-#     EXTRA_UV_SYNC_ARGS: "--extra rocm6_0"
-#     EXTRA_UV_ENV_VARS: "CUPY_INSTALL_USE_HIP=1 HCC_AMDGPU_TARGET=gfx90a"
+.build_extra_rocm:
+  variables:
+    # jfrog.svc.cscs.ch/dockerhub/rocm is the cached version of docker.io/rocm
+    BASE_IMAGE: jfrog.svc.cscs.ch/dockerhub/rocm/dev-ubuntu-${UBUNTU_VERSION}:${ROCM_VERSION}-complete
+    EXTRA_UV_SYNC_ARGS: "--extra rocm6_0"
+    EXTRA_UV_ENV_VARS: "CUPY_INSTALL_USE_HIP=1 HCC_AMDGPU_TARGET=gfx942 ROCM_HOME=/opt/rocm"
+    UBUNTU_VERSION: '22.04'
 
 build_cscs_gh200:
   extends:
     - .container-builder-cscs-gh200
     - .build_common
     - .build_extra_cuda
+  needs: []
 
-# .build_cscs_amd_rocm:
-#   extends:
-#     - .container-builder-cscs-zen2
-#     - .build_common
-#     - .build_extra_rocm
+build_cscs_amd_rocm:
+  extends:
+    - .container-builder-cscs-zen2
+    - .build_common
+    - .build_extra_rocm
+  needs: []
 
 # -- Test stage --
 .test_common:
   stage: test
   image: ${CSCS_REGISTRY_PATH}/public/${ARCH}/base/gt4py-ci-${PY_VERSION}:${DOCKER_TAG}
   variables:
+    TEST_VARIANTS: 'cpu'  # Extended jobs should redefine which variants (cpu, cuda12, rocm6_0) to test 
     CSCS_CUDA_MPS: 1
     SLURM_GPUS_PER_NODE: 4
     SLURM_JOB_NUM_NODES: 1
     SLURM_TIMELIMIT: 5
   parallel:
     matrix:
       - SUBPACKAGE: [cartesian]
-        VARIANT: ["internal", "dace"]
-        SUBVARIANT: ["cuda12", "cpu"]
+        VARIANT: ['internal', 'dace']
+        SUBVARIANT: ['cuda12', 'rocm6_0', 'cpu']
         PY_VERSION: *test_python_versions
       - SUBPACKAGE: eve
         PY_VERSION: *test_python_versions
       - SUBPACKAGE: next
-        VARIANT: ["internal", "dace"]
-        SUBVARIANT: ["cuda12", "cpu"]
+        VARIANT: ['internal', 'dace']
+        SUBVARIANT: ['cuda12', 'rocm6_0', 'cpu']
         DETAIL: ["nomesh", "atlas"]
         PY_VERSION: *test_python_versions
       - SUBPACKAGE: [storage]
-        VARIANT: ["cuda12", "cpu"]
+        VARIANT: ['cuda12', 'rocm6_0', 'cpu']
         PY_VERSION: *test_python_versions
-  rules:
+  rules: &exclude_variants_rules
+    - if: '($SUBVARIANT == "cpu" || ($SUBPACKAGE == "storage" && $VARIANT == "cpu")) && ($TEST_VARIANTS !~ /(^|\s)cpu(\s|$)/)'
+      when: never
+    - if: '($SUBVARIANT == "cuda12" || ($SUBPACKAGE == "storage" && $VARIANT == "cuda12")) && ($TEST_VARIANTS !~ /(^|\s)cuda12(\s|$)/)'
+      when: never
+    - if: '($SUBVARIANT == "rocm6_0" || ($SUBPACKAGE == "storage" && $VARIANT == "rocm6_0")) && ($TEST_VARIANTS !~ /(^|\s)rocm6_0(\s|$)/)'
+      when: never
```

Comment:
I understand your idea, but to me it feels too fancy (unless you plan for 5 more device kinds) with the complicated semantics of how overriding works (at least for non-gitlab experts). But maybe it's just the general feeling of being overwhelmed when you learn a new language...